### ISSUE TITLE
Synapse 500's on syncing with an empty sync token

### ISSUE THREAD

**Issue by anoadragon453** [matrix-org/synapse#6030]
Calling `/sync` with `...&since=` will presumably give Synapse a sync token of "", which it will then try to do stuff with. This will cause an Exception to be raised somewhere and Synapse will throw a 500 Internal Server Error.

I assume there's likely a lot more cases of this around the codebase but at least this one could be fixed.

Brought up when writing https://github.com/ara4n/random/blob/master/bashtrix.sh

**Comment by richvdh** [comment]
> I assume there's likely a lot more cases of this around the codebase but at least this one could be fixed.

Yeah. I would love it if we could actually take the time to make synapse verify its parameters against the swagger specs, rather than repeatedly finding individual cases like this, and filling up the code with boilerplate.

**Comment by ulope** [comment]
The same happens when passing an otherwise invalid value (in our case `null`). Which leads to a somewhat funny: 
```
ValueError: invalid literal for int() with base 10: 'ull'
```
in 
```
  File "/synapse-venv/lib/python3.7/site-packages/synapse/types.py", line 347, in room_stream_id
    return int(self.room_key[1:].split("-")[-1])
```
